x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -208,6 +212,48 @@ else if (p instanceof Lookup lookup) {
         return failures;
     }
 
+    private static void checkMetadataScoreNameReserved(LogicalPlan p, Set<Failure> failures) {
+        // _score can only be set as metadata attribute
+        if (p.inputSet().stream().anyMatch(a -> MetadataAttribute.SCORE.equals(a.name()) && (a instanceof MetadataAttribute) == false)) {
+            failures.add(fail(p, "`" + MetadataAttribute.SCORE + "` is a reserved METADATA attribute"));
+        }
+    }
```

Comment:
> making _score a "reserved" metadata attribute

We don't have that with any other metadata attribute -- i don't know if/why we should have this for _score.

Reply:
I think it is correct to allow manipulation of `_id`, `_index` & co because they have static values that are not being altered implicitly during query execution, `_score` is slightly different because it's currently "populated" only as a side effect of full text operations. Allowing things like `eval _score = ...` would very likely result in confusing behaviors as the user explicitly states what the `_score` column should contain, only to possibly see it overridden by full text clauses. 
I've added a task to specifically address this in #116599.

Reply:
> only to possibly see it overridden by full text clauses.

But this wouldn't happen, right? If I write `| EVAL _score = _score * 2 | WHERE field:"foo"`, the match operator wouldn't override anything, it'd just change the input value to `EVAL`.

But anyways, I'm just advising for consistency of treating all metadata fields similarly, since I still don't see a strong case this restriction, but if you think it should go in, feel free to keep it.